samples/kfp-user-guide/README.md

```diff
@@ -79,13 +79,37 @@ Now, we should able to see the pipeline is being uploaded to the **Pipelines** p
 
 ### 2. Upload Pipelines Using the `kfp_tekton.TektonClient` in Python
 
-To upload the pipelines using Python, run the below code block inside the Python session. The below code block shows how to upload different versions of the pipeline using the Python client.
+To begin, we first need to declare our TektonClient:
+- For single user:
 ```python
-import kfp_tekton
-import os
-
+from kfp_tekton import TektonClient
 # **Important**: Replace None to the KFP endpoint if the python session is not running on the Kubeflow cluster.
 host = None
+KUBEFLOW_PROFILE_NAME = None
+client = TektonClient(host=host)
+```
+
+- For multi tenant:
+1. `KUBEFLOW_PUBLIC_ENDPOINT_URL` - Kubeflow public endpoint URL. 
+2. `SESSION_COOKIE` - A session cookie starts with authservice_session=. You can obtain it from your browser after authenticated from Kubeflow UI. Notice that this session cookie expires in 24 hours, so you need to obtain it again after cookie expired.
```

Comment:
```suggestion
2. `SESSION_COOKIE` - A session cookie starts with authservice_session=. You can obtain it from your browser after authenticating with Kubeflow UI. Notice that this session cookie expires in 24 hours, so you need to obtain it again after cookie expired.
```

Reply:
Any other alternative way?

Reply:
For DEX this is the only way since they don't have programmatic authentication. The workaround in KFServing and kfctl is passing the session cookie as part of the request which is what we did here. https://github.com/kubeflow/kfctl/issues/140

For google IAP they have an IAP client for their authentication service since they are not using DEX.